### DISCUSSION TITLE
Bug with nextjs + rewrite url

### DISCUSSION THREAD

**Discussion by plHuan455** [pmndrs/jotai#2689] [Bug report]
### Bug Description

When using Jotai with Next.js 14.2.3, even though I have written a different value to an atom, when accessing the rewritten path, I still only see the old value with useHydrateAtoms.
this is my rewrite configs:
```javascript
async rewrites() {
     return [
        {
            source: `/:path/app/rewrite-shop/create`,
            destination: "/:path/app/shops/create",
        }
     ]
},
 ```

**Comment by dai-shi** [comment]
If rewrite happens on the client, you probably need to use `useEffect` to update atom values.

**Reply by DavidJFelix** [reply]
So I think rewrites is part of next's serverside routing as well, but I'm not sure if it's **exclusively** serverside. I think it would be helpful to see a minimal reproduction because I think rewrites behavior varies differs based on nextjs backend. I know the vercel variant runs rewrites on the workers routing layer.

**Comment by dai-shi** [comment]
@plHuan455 Please provide a reproduction link, preferably with stackblitz.
